framework/src/utils/InputParameters.C

```diff
@@ -388,15 +388,15 @@ InputParameters::declareControllable(const std::string & input_names,
 }
 
 bool
-InputParameters::isControllable(const std::string & name)
+InputParameters::isControllable(const std::string & name) const
 {
-  return _params.count(name) > 0 && _params[name]._controllable;
+  return _params.count(name) > 0 && _params.at(name)._controllable;
 }
 
 const std::set<ExecFlagType> &
-InputParameters::getControllableExecuteOnTypes(const std::string & name)
+InputParameters::getControllableExecuteOnTypes(const std::string & name) const
 {
-  return _params[name]._controllable_flags;
+  return _params.at(name)._controllable_flags;
```

Comment:
```suggestion
  const auto it = _params.find(name);
  if (it != _params.end())
    return *it;
  const static std::set<ExecFlagType> empty_types;
  return empty_types;
```

either this or an error. the unintentional insertion we had here before is very bad :(

Reply:
Yeah, I was hoping a comment like this ;-)

Reply:
I think it is possibly better to throw an error if the key is not found.

Reply:
If so, then we should probably do the same for `isControllable`

Reply:
That has the check whether the parameter is valid. Hold on, actually `Metadata` has a special implementation of `at`, so my original code is ok.